keycloak/keycloak-admin-ui
# User Federation: populate all LDAP settings

## Motivation
Related JIRA issues:
https://issues.redhat.com/browse/APPDUX-655

Design spec:
https://marvelapp.com/prototype/ecjc94b/screen/72098368

## Brief Description
Fetches and displays settings data for a specific individual LDAP provider card that is clicked in the user federation card view. 

This PR does not cover Save nor the showing/hiding of UI components based on selections. These will be handled in future PRs. This PR only covers pulling in the data and getting to the settings page from the cards.

## Verification Steps
1. Click an LDAP card in the user fed card view.
2. Verify that all settings match those that have been configured for that LDAP provider (presumably, the ones configured and saved in the old admin UI). There should be 7 sections: **General options**, **Connection and authentication settings**, **LDAP searching and updating**, **Synchronization settings**, **Kerberos integration**, **Cache settings**, and **Advanced settings**.
3. Change the values in the old admin UI and verify that the changes are reflected in the new UI.
4. Click a Kerberos card in the user fed card view.
5. Verify that the kerberos-specific settings are displayed (**Required settings** and **Cache settings**).

## Checklist:

## Additional Notes
Screen caps:
![ldap-cards](https://user-images.githubusercontent.com/39063664/102140663-976ebf00-3e2d-11eb-88d6-dbbda60ad33e.png)

![ldap-settings-1](https://user-images.githubusercontent.com/39063664/102140669-9b9adc80-3e2d-11eb-882f-fb9815ce13e0.png)

![ldap-settings-2](https://user-images.githubusercontent.com/39063664/102140674-9dfd3680-3e2d-11eb-8927-a428dccd8d07.png)

![ldap-settings-3](https://user-images.githubusercontent.com/39063664/102140681-a0f82700-3e2d-11eb-84f2-582a0aaac77e.png)

![ldap-settings-4](https://user-images.githubusercontent.com/39063664/102140692-a35a8100-3e2d-11eb-8476-05801cef7fb1.png)
